packages/fxa-settings/src/pages/InlineTotpSetup/index.tsx

```diff
@@ -0,0 +1,288 @@
+                >
+                  these authentication apps
+                </LinkExternal>
+                .
+              </p>
+            </FtlMsg>
+            <button
+              type="submit"
+              className="cta-primary cta-xl w-full m-2 my-6"
+              onClick={() => {
+                // TODO: Add in any further functionality here
+                setShowIntro(false);
+              }}
+            >
+              <FtlMsg id="inline-totp-setup-continue-button">Continue</FtlMsg>
+            </button>
+            <button type="button" className="link-blue text-sm">
+              <FtlMsg id="inline-totp-setup-cancel-setup-button">
+                Cancel setup
+              </FtlMsg>
+            </button>
+          </section>
+        </>
+      )}
+      {!showIntro && (
+        <>
+          {showQR ? (
+            <CardHeader
+              headingText="Scan authentication code"
+              headingWithCustomServiceFtlId="inline-totp-setup-show-qr-custom-service-header"
+              headingWithDefaultServiceFtlId="inline-totp-setup-show-qr-default-service-header"
+              {...{ serviceName }}
+            />
+          ) : (
+            <CardHeader
+              headingText="Enter code manually"
+              headingWithCustomServiceFtlId="inline-totp-setup-no-qr-custom-service-header"
+              headingWithDefaultServiceFtlId="inline-totp-setup-no-qr-default-service-header"
+              {...{ serviceName }}
+            />
+          )}
+          <section>
+            <form noValidate>
+              <div id="totp" className="totp-details">
+                {showQR ? (
+                  <>
+                    <FtlMsg
+                      id="inline-totp-setup-use-qr-or-enter-key-instructions"
+                      elems={{
+                        toggleToManualModeButton: (
+                          <button
+                            type="button"
+                            onClick={() => {
+                              setShowQR(false);
+                            }}
+                            className="link-blue inline"
+                          >
+                            Can’t scan code?
+                          </button>
+                        ),
+                      }}
+                    >
+                      <p className="text-sm mb-4">
+                        Scan the QR code in your authentication app and then
+                        enter the security code it provides.{' '}
+                        <button
+                          type="button"
+                          onClick={() => {
+                            setShowQR(false);
+                          }}
+                          className="link-blue inline"
+                        >
+                          Can’t scan code?
+                        </button>
+                      </p>
+                    </FtlMsg>
+                    <div>
+                      <img
```

Comment:
I wonder if this needs a width/height class on it to prevent jank when it loads. Might could add a comment to the functional ticket for this unless you need to push here again.

Reply:
Agreed! Added to the ticket